### PR TITLE
Add version info in docs

### LINKED ISSUE #252: Move API reference in docs out of About section

### PR DESCRIPTION
**Pull request by nvaytet**
- Add a 'current version' footer
- Move api reference page out of the About section
- Add note on where to find older docs in the about page

Fixes #252

### CODE REVIEW

File: docs/_templates/doc_version.html [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<!-- This will display the version of the docs -->
+Current {{project }} version: {{ version }} (<a href="https://github.com/scipp/plopp/releases">older versions</a>).
```
Comment (jl-wynen) [inline docs/_templates/doc_version.html:2]:
```suggestion
Current {{project }} version: {{ version }} (<a href="https://github.com/scipp/{{plopp}}/releases">older versions</a>).
```
?

File: docs/about/about.md [hunk 2]
```diff
@@ -8,6 +8,11 @@ Plopp is an open source project by the [European Spallation Source ERIC](https:/
 
 Plopp is available as open source under the [BSD-3 license](https://opensource.org/licenses/BSD-3-Clause).
 
+## Older versions of the documentation
+
+Older versions of the documentation pages can be found under the assets of each [release](https://github.com/scipp/plopp/releases).
```
Comment (jl-wynen) [inline docs/about/about.md:13]: Also needs to be adapted when moving to the template.

File: docs/_templates/doc_version.html (hunk 1 above)
Comment (nvaytet) [reply docs/_templates/doc_version.html:2]: Did you mean `{{project}}`?

Comment (jl-wynen) [reply docs/_templates/doc_version.html]: Yes :)

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**